src/modules/Dashboard/features/RespondentData/RespondentDataReview/hooks/useActivityAnswersAndAssessment/useActivityAnswersAndAssessment.ts

```diff
@@ -133,7 +134,31 @@ export const useActivityAnswersAndAssessment = ({
       setIsLoading(true);
       if (flowId && submitId) {
         await getFlowAnswers({ appletId, flowId, submitId });
-        // TODO: add flow assessment implementation after backend is ready (BE task M2-6584)
+        const {
+          data: { result: assessmentResult },
+        } = await getFlowAssessmentApi({ appletId, submitId });
+
+        const { reviewerPublicKey, itemsLast, versions, items, ...assessmentData } =
```

Comment:
Could you please move the repeating code (lines 141-161 and 169-189) to a function and reuse it in both places?

Reply:
the PR is in progress